Detectors/TRD/calibration/include/TRDCalibration/DCSProcessor.h

```diff
@@ -64,33 +64,64 @@ class DCSProcessor
   bool updateCurrentsDPsCCDB();
   bool updateEnvDPsCCDB();
   bool updateRunDPsCCDB();
+  // LB: new DPs for Fed
+  bool updateFedChamberStatusDPsCCDB();
+  bool updateFedCFGtagDPsCCDB();
+  bool updateFedEnvTempDPsCCDB();
 
   // signal that the CCDB object for the voltages should be updated due to change exceeding threshold
   bool shouldUpdateVoltages() const { return mShouldUpdateVoltages; }
   bool shouldUpdateRun() const { return mShouldUpdateRun; }
+  // LB: Only update ChamberStatus/CFGtag if both conditions are met (complete DPs and new run)
+  bool shouldUpdateFedChamberStatus() const { return mFedChamberStatusCompleteDPs && mFirstRunEntryForFedChamberStatusUpdate; }
+  bool shouldUpdateFedCFGtag() const { return mFedCFGtagCompleteDPs && mFirstRunEntryForFedCFGtagUpdate; }
+  // LB: Env DPs have no alias pattern, processor uses this function to identify if alias is Env
+  bool isAliasFromEnvDP(const char* dpalias) const
+  {
+    std::vector<std::string> envaliases = {"CavernTemperature", "temperature_P2_external", "AtmosPressure", "UXC2Humidity"};
+    for (const auto& envalias : envaliases) {
+      if (std::strstr(dpalias, envalias.c_str()) != nullptr) {
+        return true;
+      }
+    }
+    return false;
+  }
 
   // allow access to the CCDB objects from DPL processor
   CcdbObjectInfo& getccdbGasDPsInfo() { return mCcdbGasDPsInfo; }
   CcdbObjectInfo& getccdbVoltagesDPsInfo() { return mCcdbVoltagesDPsInfo; }
   CcdbObjectInfo& getccdbCurrentsDPsInfo() { return mCcdbCurrentsDPsInfo; }
   CcdbObjectInfo& getccdbEnvDPsInfo() { return mCcdbEnvDPsInfo; }
   CcdbObjectInfo& getccdbRunDPsInfo() { return mCcdbRunDPsInfo; }
+  CcdbObjectInfo& getccdbFedChamberStatusDPsInfo() { return mCcdbFedChamberStatusDPsInfo; }
+  CcdbObjectInfo& getccdbFedCFGtagDPsInfo() { return mCcdbFedCFGtagDPsInfo; }
+  CcdbObjectInfo& getccdbFedEnvTempDPsInfo() { return mCcdbFedEnvTempDPsInfo; }
   const std::unordered_map<DPID, TRDDCSMinMaxMeanInfo>& getTRDGasDPsInfo() const { return mTRDDCSGas; }
   const std::unordered_map<DPID, float>& getTRDVoltagesDPsInfo() const { return mTRDDCSVoltages; }
   const std::unordered_map<DPID, TRDDCSMinMaxMeanInfo>& getTRDCurrentsDPsInfo() const { return mTRDDCSCurrents; }
   const std::unordered_map<DPID, TRDDCSMinMaxMeanInfo>& getTRDEnvDPsInfo() const { return mTRDDCSEnv; }
   const std::unordered_map<DPID, int>& getTRDRunDPsInfo() const { return mTRDDCSRun; }
+  const std::unordered_map<DPID, int>& getTRDFedChamberStatusDPsInfo() const { return mTRDDCSFedChamberStatus; }
```

Comment:
```suggestion
  const std::array<constants::MAXCHAMBER, int>& getTRDFedChamberStatusDPsInfo() const { return mTRDDCSFedChamberStatus; }
```
Probably this data type will make it easier to use the chamber status in QC and in our simulation. Then we don't need the DPID definition everywhere. The same might be a good idea for `mTRDDCSFedCFGtag`.
The other DPs will likely only be used in macros, therefore I had started with the map data type there.